Makefile

```
@@ -44,10 +44,24 @@ LDFLAGS := -X github.com//open-telemetry/opentelemetry-ebpf-profiler/vc.version=
 	-X github.com/open-telemetry/opentelemetry-ebpf-profiler/vc.buildTimestamp=$(BUILD_TIMESTAMP) \
 	-extldflags=-static
 
-GO_FLAGS := -buildvcs=false -ldflags="$(LDFLAGS)" -tags osusergo,netgo
+# Valid values: release, debug
+BUILD_TYPE ?= release
+
+GO_TAGS := osusergo,netgo
+DEBUG_FLAGS := 
+
+ifeq ($(BUILD_TYPE),debug)
+	GO_TAGS = osusergo,netgo,debugtracer
+	DEBUG_FLAGS += debug
+endif
+
+GO_FLAGS := -buildvcs=false -ldflags="$(LDFLAGS)" -tags $(GO_TAGS)
```

I would simplify a little and remove the recursive `MAKE` call by splitting `GO_TAGS` out of `GO_FLAGS` (see my other suggestions). Will also need to update the two places that GO_FLAGS is used and add `-tags $(GO_TAGS)`

(I know that the `Makefile` in support/ebpf is using the same pattern, and we've been planning a total build system overhaul for some time, but that's for another day)

```suggestion
GO_FLAGS := -buildvcs=false -ldflags="$(LDFLAGS)"
```